### PR TITLE
Increase FlyCircle ticks to 1 when idling

### PR DESCRIPTION
**Pull request by tovl**
fixes issue #16134.

FlyCircle would run indefinitely and queued activities don't cancel a running activity, so changed FlyCircle to run 1 tick instead of infinite ticks.

### CODE REVIEW

**Comment by pchote** [comment]
I'm not a big fan of creating so many throw away activities - how about adding a "cancel if something else is queued" flag to `FlyCircle` instead?

**Comment by tovl** [comment]
Done.

**Review by ghost — COMMENTED** [review]
This causes a freeze for me when

- build 1 airfield
- build 2 yaks
- order a yak to force fire somewhere, then hit F so it reloads
- hit F on the other yak

@tovl btw, if you didn't notice, there is an IRC channel at irc://chat.freenode.net/openra where you can talk about your PRs with the devs. you can also join via the web interface at http://webchat.freenode.net/?channels=openra (but you need to register a nickname) :)

**Comment by tovl** [comment]
Regression is fixed. It seems a line that was intended to prevent an infinite loop, now caused an infinite loop.

>there is an IRC channel

Thanks for the tip!

**Review by ghost — COMMENTED** [review]
> Regression is fixed. It seems a line that was intended to prevent an infinite loop, now caused an infinite loop.

Unfortunately not :(

It now freezes when you attack something with multiple yaks and issue "F" on all selected yaks (1 airfield).

This will require more changes.

Edit: Your last push didn't fix the first issue, this appears to be the same bug as before.

**Review by ghost — COMMENTED** [review]
https://github.com/OpenRA/OpenRA/commit/64ba861d11bfe655f59510642ba52aa0821535ec has fixed #16134 and I can't find other issues here :+1: 

Edit: Still valid for https://github.com/OpenRA/OpenRA/commit/850f9376a8f989e50861e74db54f9bef30e93a36.

File: OpenRA.Mods.Common/Activities/Air/FlyCircle.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ public FlyCircle(Actor self, int ticks = -1, int turnSpeedOverride = -1)
 
 		public override Activity Tick(Actor self)
 		{
+			if (NextActivity != null && remainingTicks < 0)
```
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Air/FlyCircle.cs]: If you make this `remainingTicks <= 0` then you can drop the redundant `else if (remainingTicks == 0)` line below.

**Review by obrakmann — APPROVED** [review]
lgtm :+1:

**Review by pchote — APPROVED** [review]

**Pull request merged**